### PR TITLE
Refs [TASK][34939] Check internet

### PR DESCRIPTION
**Pull request by manhhung23399**
## Related Tickets
- [#Task 34939: Check internet](https://edu-redmine.sun-asterisk.vn/issues/34939)
## WHAT
- linstener internet change
## Review Checklist

Category | View Point | Description | Expected Reviewer Answer | Self review | Reviewer2 (name)
--- | --- | --- | --- | --- | ---
Conventions | Does the code follow Sun* coding style and coding conventions? | https://github.com/framgia/coding-standards/tree/master/eng/android | YES |<li>- [ ] yes</li>|<li>- [ ] yes</li>  
Redmine | Does the ticket follow Sun* Redmine working process?  | https://github.com/framgia/Training-Guideline/blob/master/WorkingProcess/redmine/redmine.md| YES |<li>- [ ] yes</li>|<li>- [ ] yes</li>  
Documentation | Is there any incomplete code? If so, should it be removed or flagged with a suitable marker like ‘TODO’? |  | YES |<li>- [ ] yes</li>|<li>- [ ] yes</li>  
## Notes (Optional)
*(Impacted Areas in Application(List features, api, models or services that this PR will affect))*

*(List gem, library third party add new)*

*(Other notes)*

### CODE REVIEW

File: app/src/main/java/com/example/moviews/screen/error/ErrorFragment.kt [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+
+    private fun checkInternet(): Boolean {
+        val connectivityManager =
+            activity?.getSystemService(Context.CONNECTIVITY_SERVICE) as ConnectivityManager
+        val networkInfo = connectivityManager.activeNetworkInfo
+        return networkInfo != null && networkInfo.isConnected
+    }
+
+    companion object {
+        fun getInstance(): ErrorFragment = ErrorFragment()
```
Comment (tuannm-1276) [inline app/src/main/java/com/example/moviews/screen/error/ErrorFragment.kt]: sai rồi

File: app/src/main/java/com/example/moviews/screen/error/ErrorFragment.kt [hunk 2]
```diff
@@ -0,0 +1,40 @@
+package com.example.moviews.screen.error
+
+import android.content.Context
+import android.net.ConnectivityManager
+import com.example.moviews.R
+import com.example.moviews.base.BaseFragment
+import com.example.moviews.screen.home.HomeFragment
+import com.example.moviews.utils.addFragment
+import kotlinx.android.synthetic.main.fragment_error.*
+
```
Comment (tuannm-1276) [inline app/src/main/java/com/example/moviews/screen/error/ErrorFragment.kt:11]: blank

File: app/src/main/java/com/example/moviews/screen/error/ErrorFragment.kt [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+    override fun initViews() {
+    }
+
+    override fun initData() {
+    }
+
+    override fun initEvents() {
+        chipRetry.setOnClickListener {
+            if (checkInternet()) {
+                addFragment(HomeFragment.newInstance())
```
Comment (tuannm-1276) [inline app/src/main/java/com/example/moviews/screen/error/ErrorFragment.kt]: add thằng Home vào đâu đây em?

Comment (manhhung23399) [reply app/src/main/java/com/example/moviews/screen/error/ErrorFragment.kt]: add vào fragment container ạ, e có viết FragmentExtension rồi a

**Review by datdt-1935 — APPROVED** [review]

**Pull request closed without merging**